### PR TITLE
[PHI] move huber_loss/huber_loss_grad xpu kernel to phi

### PR DESCRIPTION
**Pull request by pangyoki**
### PR types

Others

### PR changes

OPs

### Describe

move huber_loss/huber_loss_grad xpu kernel to phi.

### CODE REVIEW

File: paddle/phi/kernels/xpu/huber_loss_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  int r = xpu::huber_loss<T>(dev_ctx.x_context(),
+                             in0_data,
+                             in1_data,
+                             residual_data,
+                             out_data,
+                             input.numel(),
+                             1,
+                             delta);
+  PADDLE_ENFORCE_EQ(r,
```
Comment (helen88) [inline paddle/phi/kernels/xpu/huber_loss_kernel.cc]: PADDLE_ENFORCE_XDNN_SUCCESS 用这个宏

File: paddle/phi/kernels/xpu/huber_loss_grad_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  auto residual_data = residual.data<T>();
+  int r = xpu::huber_loss_grad<T>(dev_ctx.x_context(),
+                                  residual_data,
+                                  out_grad_data,
+                                  input_grad_data,
+                                  label_grad_data,
+                                  out_grad.numel(),
+                                  1,
+                                  delta);
+  PADDLE_ENFORCE_EQ(
```
Comment (helen88) [inline paddle/phi/kernels/xpu/huber_loss_grad_kernel.cc]: PADDLE_ENFORCE_XDNN_SUCCESS 请用这个宏

File: paddle/phi/kernels/xpu/huber_loss_kernel.cc (hunk 1 above)
Comment (pangyoki) [reply paddle/phi/kernels/xpu/huber_loss_kernel.cc]: done, thx

File: paddle/phi/kernels/xpu/huber_loss_grad_kernel.cc (hunk 2 above)
Comment (pangyoki) [reply paddle/phi/kernels/xpu/huber_loss_grad_kernel.cc]: done, thx

**Review by chenwhql — APPROVED** [review]

**Pull request merged**